Makefile

```diff
@@ -0,0 +1,505 @@
+  PLATFORM_DIR       := darwin-x86_64
+  LIBRARY_EXTENSION  := dylib
+else
+  $(error Unsupported host platform $(UNAME_S))
+endif
+
+# Setup Android toolchain
+# Preferred NDK version
+ANDROID_NDK_PREFER_VERSION ?= 28.1
+
+# Autodetect or use existing NDK_HOME/NDK_HOME
+ANDROID_NDK_ROOT = $(strip \
+  $(or \
+    $(ANDROID_NDK_HOME), \
+    $(NDK_HOME), \
+    $(shell [ -d "$(HOME)/Android/Sdk/ndk" ] && \
+               find "$(HOME)/Android/Sdk/ndk" -maxdepth 1 -type d \
+                    -name "$(ANDROID_NDK_PREFER_VERSION)*" \
+               | head -n1) \
+  ) \
+)
+
+
+# Build paths
+NDK_TOOLCHAIN := $(ANDROID_NDK_ROOT)/toolchains/llvm/prebuilt/$(HOST_DIR)
+NDK_BIN       := $(NDK_TOOLCHAIN)$(PLATFORM_DIR)/bin
+CLANG_RT_DIR = $(strip \
+  $(shell \
+    [ -n "$(ANDROID_NDK_ROOT)" ] && \
+    "$(NDK_BIN)/clang" --print-runtime-dir \
+      | xargs dirname | xargs dirname | xargs -I{} echo {}/linux \
+  ) \
+)
+
+
+# Export to sub‐recipes
+export PATH          := $(PATH):$(NDK_BIN)
+export ANDROID_NDK_HOME := $(ANDROID_NDK_ROOT)
+export NDK_HOME         := $(ANDROID_NDK_ROOT)
+export CLANG_RT_DIR
+
+
+# Tools
+CARGO          := cargo
+BUN            := bun
+WASM_PACK      := wasm-pack
+UNIFFI_BINDGEN := target/release/uniffi-bindgen
+
+# Default goal
+.DEFAULT_GOAL := all
+
+# Directory we store timestamps in
+STAMPS := .stamps
+TOUCH_STAMP = @mkdir -p $(STAMPS) && touch $@
+
+#-------------------------------------------------------------------------------
+# 1) Dummy force target: always out-of-date, so Cargo build commands run every time
+#-------------------------------------------------------------------------------
+.PHONY: FORCE
+FORCE:
+
+#-------------------------------------------------------------------------------
+# 2) Build artifacts via Cargo (always run Cargo; Cargo itself detects up-to-date)
+#-------------------------------------------------------------------------------
+
+# Build bindgen binary
+target/release/uniffi-bindgen: FORCE
+	cargo build --release \
+		--locked \
+		--features uniffi/cli \
+		--package core-crypto-ffi \
+		--bin uniffi-bindgen
+
+# Build the FFI library
+target/release/libcore_crypto_ffi.$(LIBRARY_EXTENSION): FORCE
+	cargo build --release \
+		--locked \
+		--package core-crypto-ffi \
+		--lib
+
+# Make aliases
+.PHONY: uniffi-bindgen release-build
+uniffi-bindgen:  target/release/uniffi-bindgen
+release-build:   target/release/libcore_crypto_ffi.$(LIBRARY_EXTENSION) ## Build core-crypto-ffi for the native arch (needed for uniffi)
+
+#-------------------------------------------------------------------------------
+# 3) Use stamp files for generators: only re-run when inputs change
+#-------------------------------------------------------------------------------
+
+# Swift bindings
+ifneq ($(UNAME_S),Darwin)
+$(STAMPS)/bindings-swift:
+	$(error Swift bindings generation is only supported on Darwin because OpenSSL can't be cross-compiled on non-Darwin systems; this is "$(UNAME_S)".)
```

Comment:
This should be a warning, not an error, because otherwise targets like `bindings` won't work at all on non-Apple platforms.